cloud/scope/powervs_cluster.go

```diff
@@ -157,3 +261,1558 @@ func (s *PowerVSClusterScope) PatchObject() error {
 func (s *PowerVSClusterScope) Close() error {
 	return s.PatchObject()
 }
+
+// Name returns the CAPI cluster name.
+func (s *PowerVSClusterScope) Name() string {
+	return s.Cluster.Name
+}
+
+// Zone returns the cluster zone.
+func (s *PowerVSClusterScope) Zone() *string {
+	return s.IBMPowerVSCluster.Spec.Zone
+}
+
+// ResourceGroup returns the cluster resource group.
+func (s *PowerVSClusterScope) ResourceGroup() *infrav1beta2.IBMPowerVSResourceReference {
+	return s.IBMPowerVSCluster.Spec.ResourceGroup
+}
+
+// InfraCluster returns the IBMPowerVS infrastructure cluster object.
+func (s *PowerVSClusterScope) InfraCluster() ClusterObject {
+	return s.IBMPowerVSCluster
+}
+
+// APIServerPort returns the APIServerPort to use when creating the ControlPlaneEndpoint.
+func (s *PowerVSClusterScope) APIServerPort() int32 {
+	if s.Cluster.Spec.ClusterNetwork != nil && s.Cluster.Spec.ClusterNetwork.APIServerPort != nil {
+		return *s.Cluster.Spec.ClusterNetwork.APIServerPort
+	}
+	return 6443
```

Comment:
can we create a constant somewhere for this port and use it?

Reply:
Created a constant for it.